### PR TITLE
Slash commands

### PR DESCRIPTION
**Pull request by andypayne**
Fix for issue #32 - Support sending control frames.

### CODE REVIEW

File: bin/wscat [hunk 1]
```diff
@@ -227,7 +228,28 @@ if (program.listen && program.connect) {
       } else {
         wsConsole.print(Console.Types.Control, 'connected (press CTRL+C to quit)', Console.Colors.Green);
         wsConsole.on('line', function line(data) {
-          ws.send(data);
+          if (program.slash && data[0] === '/') {
+            switch(data.substr(1)) {
+              case 'ping':
+                console.log('Sending ping');
```
Comment (lpinca) [inline bin/wscat]: Are these wanted?

File: bin/wscat [hunk 2]
```diff
@@ -227,7 +228,28 @@ if (program.listen && program.connect) {
...
+                ws.ping(function() { });
+                break;
+              case 'pong':
+                console.log('Sending pong');
+                ws.pong(function() { });
```
Comment (lpinca) [inline bin/wscat]: I'd declare a single noop function and use it for both ping and pong.

Comment (andypayne) [reply bin/wscat]: Done in 752a40f

File: bin/wscat (hunk 1 above)
Comment (andypayne) [reply bin/wscat]: Are you asking whether the ping command is wanted? I'm not sure what you're asking.

Comment (andypayne) [reply bin/wscat]: Ah, sorry. Those are removed now in 6ebafbb.

File: bin/wscat [hunk 3]
```diff
@@ -227,7 +228,29 @@ if (program.listen && program.connect) {
...
+          if (program.slash && data[0] === '/') {
+            switch(data.split(/\s+/)[0].substr(1)) {
+              case 'ping':
+                ws.ping(noop);
+                break;
+              case 'pong':
+                ws.pong(noop);
+                break;
+              case 'close':
+                var toks = data.split(/\s+/);
```
Comment (lpinca) [inline bin/wscat]: Can we use this only once? I think we can get the tokens before the `switch` statement and reuse the result here if necessary.

Also you can use `const` and `let`. This no longer works in Node.js < 4 as it depends on `ws@5`.

Thanks.

File: bin/wscat [hunk 4]
```diff
@@ -227,7 +228,29 @@ if (program.listen && program.connect) {
...
+                break;
+              case 'pong':
+                ws.pong(noop);
+                break;
+              case 'close':
+                let closeStatusCode = 1000;
+                if (toks.length === 2) {
+                  closeStatusCode = parseInt(toks[1]);
+                }
+                ws.close(closeStatusCode);
```
Comment (lpinca) [inline bin/wscat]: Should we also allow the reason now that we are at it?

File: bin/wscat (hunk 3 above)
Comment (andypayne) [reply bin/wscat]: That's done. That's now the only area where const and let are used. The whole file needs some cleanup to bring it up to date if going that route.

File: bin/wscat (hunk 4 above)
Comment (andypayne) [reply bin/wscat]: That's added now as well.

**Comment by lpinca** [comment]
@nwwells do you want to take a look?

**Comment by nwwells** [comment]
Looking now...

**Comment by lpinca** [comment]
Thank you.

**Pull request merged**